models/intermediate/int_zendesk__schedule_spine.sql

```diff
@@ -90,21 +95,224 @@ with timezone as (
         schedule.end_time - coalesce(split_timezones.offset_minutes, 0) as end_time_utc,
 
         -- we'll use these to determine which schedule version to associate tickets with
-        split_timezones.valid_from,
-        split_timezones.valid_until
+        cast(split_timezones.valid_from as {{ dbt.type_timestamp() }}) as valid_from,
+        cast(split_timezones.valid_until as {{ dbt.type_timestamp() }}) as valid_until
 
     from schedule
     left join split_timezones
         on split_timezones.time_zone = schedule.time_zone
 
-), final as (
+-- Now we need take holiday's into consideration and perform the following transformations to account for Holidays in existing schedules
+), holiday_start_end_times as (
 
     select 
+        calculate_schedules.*,
+        schedule_holiday.holiday_name,
+        schedule_holiday.holiday_start_date_at,
+        cast({{ dbt.dateadd("second", "86399", "schedule_holiday.holiday_end_date_at") }} as {{ dbt.type_timestamp() }}) as holiday_end_date_at,
```

Comment:
Correct, since the holiday start and end date are the same day, I want to capture the actual start and end times. The dbt.dateadd ensures I am capturing the very last second of the day. This way I have a window of when the holiday took place.